hnn_core/network.py

```diff
@@ -730,28 +595,30 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
         pos = [self.pos_dict['origin'] for _ in src_gid_ran]
         self._add_cell_type(name, pos)
 
-        # Update connectivity_list
-        nc_dict = {
-            'A_delay': self.delay,
-            'threshold': self.threshold,
-        }
         receptors = ['ampa', 'nmda']
         if drive['type'] == 'gaussian':
             receptors = ['ampa']
+
+        src_range = self.gid_ranges[_long_name(drive['name'])]
         # conn-parameters are for each target cell type
         for target_cell, drive_conn in drive['conn'].items():
+            target_range = self.gid_ranges[_long_name(target_cell)]
             for receptor in receptors:
                 if len(drive_conn[receptor]) > 0:
-                    nc_dict['lamtha'] = drive_conn[
-                        receptor]['lamtha']
-                    nc_dict['A_delay'] = drive_conn[
-                        receptor]['A_delay']
-                    nc_dict['A_weight'] = drive_conn[
-                        receptor]['A_weight']
-                    loc = drive_conn['location']
-                    self._all_to_all_connect(
-                        drive['name'], target_cell, loc, receptor,
-                        deepcopy(nc_dict), unique=drive['cell_specific'])
+                    if drive['cell_specific']:
```

Comment:
I noticed that you add tests in the end. Ideally, you want to start by writing the test (not example), and then work your way to make the test pass. It's a much faster way to develop and debug software. Look up "test driven development"

Reply:
Thanks for this pointer, I'll definitively start incorporating that more. Conveniently for this specific section it can be handled through `test_drives.py` since that's the only place the unique option comes into play. Since it seems weird to leave in the user API of `add_connection` I'm just getting rid of it.